CodeJam.Main.Tests/IO/TempDataTests.cs

```diff
@@ -98,6 +112,46 @@ public void TestDirectoryNestedContent()
 			Assert.IsFalse(File.Exists(nestedFile), "File should NOT exist");
 		}
 
+		[Test]
+		public void TestRetryPolicy()
+		{
+			int disposeCallCount = 0;
+			try
+			{
+				Configuration.SetTempDataRetryCallback(
+					a =>
+					{
+						Interlocked.Increment(ref disposeCallCount);
+						a();
+					});
+
+				string dirPath;
+				string nestedFile;
+				using (var dir = TempData.CreateDirectory())
+				{
+					dirPath = dir.Path;
+					nestedFile = Path.Combine(dirPath, "test.tmp");
+
+					using (File.Create(nestedFile))
+					{
+						Assert.Throws<IOException>(() => dir.Dispose());
+						Assert.AreEqual(disposeCallCount, 1);
+					}
+
+					Assert.DoesNotThrow(() => dir.Dispose());
+					Assert.AreEqual(disposeCallCount, 2);
+
+				}
+				Assert.IsFalse(Directory.Exists(dirPath), "Directory should NOT exist");
```

Comment:
Use the better constraint ;)

https://github.com/nunit/docs/wiki/FileOrDirectoryExistsConstraint

Reply:
Well, there's almost half handred Exists() calls in the file and I'm not going to rewrite them all:)

Second, I'd prefer to keep direct framework calls. Proposed one may become controversial very quickly. As example, 
`Assert.That(somevar, Does.NotExists)`.

Reply:
What is the problem ?

Assert.That(foolder, Does.Not.Exist)


And you can combine

Assert.That(foolder, Is.Not.Null.And.Does.Not.Exist)

I’ll check how we can replace all exists with this constraint